src/redistributors/morpho/Distributor.sol

```diff
@@ -4,91 +4,135 @@ pragma solidity 0.8.23;
+        _setShares(vault_, AccountFundsDistribution._getTotalShares(vault_));
+        AccountFundsDistribution._distribute(vault_, _updateLastCumulativeDistribution(vault_));
     }
 
-    /// @notice Allow `msg.sender` to claim their WrappedM yield for their deposited shares of a Morpho Vault.
+    /// @inheritdoc IDistributor
     function claim(address vault_, address recipient_) external returns (uint256 claimed_) {
         distribute(vault_);
 
+        // NOTE: Do not need ot check result of transfer as the behavior of WrappedM is known to revert on failure.
         IWrappedMLike(wrappedMToken).transfer(recipient_, claimed_ = _updateClaim(vault_, msg.sender));
 
         _lastWrappedMBalance = IWrappedMLike(wrappedMToken).balanceOf(address(this));
     }
 
-    /// @notice Returns the WrappedM yield claimable for an account's deposited shares of a Morpho Vault.
+    /* ============ View/Pure Functions ============ */
+
+    /// @inheritdoc IDistributor
     function getClaimable(address vault_, address account_) public view returns (uint256 claimable_) {
         // NOTE: Should/could never be negative.
-        return _getCumulativeDistribution(vault_, account_) - _claims[vault_][account_];
+        return AccountFundsDistribution._getCumulativeDistribution(vault_, account_) - _claims[vault_][account_];
     }
 
-    /// @notice Returns an account's deposited shares of a Morpho Vault.
-    function getDeposits(address vault_, address account_) external view returns (uint256 shares_) {
-        return _getShares(vault_, account_);
+    /// @inheritdoc IDistributor
+    function getShares(address vault_, address account_) external view returns (uint256 shares_) {
+        return AccountFundsDistribution._getShares(vault_, account_);
     }
 
-    /// @notice Returns an account's shares of distribution of a vault's distribution.
-    function getShares(address vault_, address account_) external view returns (uint256 shares_) {
-        return _getShares(vault_, account_);
+    /// @inheritdoc IDistributor
+    function getShares(address vault_) external view returns (uint256 shares_) {
+        return VaultFundsDistribution._getShares(vault_);
+    }
+
+    /// @inheritdoc IDistributor
+    function getTotalShares(address vault_) external view returns (uint256 totalShares_) {
+        return AccountFundsDistribution._getTotalShares(vault_);
+    }
+
+    /// @inheritdoc IDistributor
+    function getTotalShares() external view returns (uint256 totalShares_) {
+        return VaultFundsDistribution._getTotalShares();
+    }
+
+    /* ============ Internal Interactive Functions ============ */
+
+    /// @notice Updates the shares of a vault, first distribute any new WrappedM across all vaults.
+    function _setShares(address vault_, uint256 totalDeposited_) internal {
+        uint256 distributable_ = _updateLastWrappedMBalance();
+        uint256 vaultShares_ = _convertToAssets(vault_, totalDeposited_);
```

Comment:
The naming is a bit confusing here.

- `totalDeposited_` is the total amount of Distributor shares attributed to the `vault_` plus or minus the amount of Morpho Vault shares being deposited or withdrawn from the Distributor.
- `_getVaultTotalAssets(vault_)` is the total amount of assets (i.e. wM) held by Morpho across all markets.
- `IMorphoVaultLike(vault_).totalSupply()` is the total amount of Morpho Vault shares.
- `vaultShares_` is actually the amount of assets (i.e. wM) being staked/unstaked into/from the distributor?

So maybe the variable should be named `vaultAssets_`?

Reply:
- `totalDeposited_` is actually the total vault shares deposited by accounts, not Distributor shares attributed to the `vault_` in `VaultFundsDistribution`. In light of the confusion, I have renamed it to `totalVaultDeposits_`, and added more natspec.
- `vaultShares_` is supposed to be the new Distributor shares attributed to the `vault_`, which in fact is the value of the assets this vault has brought to the table. To simplify, it has been renamed to `vaultAssets_` as you suggested.